components/count-badge/count-badge.js

```diff
@@ -0,0 +1,136 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class CountBadge extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * The number to be displayed on the badge. If badge-type is "notification", the number will be truncated to two digits ("99+").
+			 */
+			number: {
+				type: Number,
+				reflect: true,
+				attribute: 'number'
+			},
+			/**
+			 * The size of the badge. Valid options are "small" and "large". Defaults to "small".
+			 */
+			size: {
+				type: String,
+				reflect: true,
+				attribute: 'size'
+			},
+			/**
+			 * The type of the badge. Valid options are "notification" and "count". Defaults to "count".
+			 */
+			type: {
+				type: String,
+				reflect: true,
+				attribute: 'type'
+			},
+			/**
+			 * Optionally specify a digit limit, after which numbers are truncated. Defaults to two for "notification" type and no limit for "count" type.
+			 */
+			maxDigits: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-digits'
+			},
+			/**
+			 * Optionally choose to not render the count badge when the number is zero. Defaults to false.
+			 */
+			hideZero: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'hide-zero'
+			},
+		};
+	}
+
+	static get styles() {
+		return [ css`
+		:host([hidden]) {
+			display: none;
+		}
+
+		.d2l-count-badge-number {
+			font-weight: bold;
+		}
+
+		:host([type="notification"]) .d2l-count-badge-number {
+			color: white;
+		}
+
+		:host([type="count"]) .d2l-count-badge-number {
+			color: var(--d2l-color-tungsten);
+		}
+
+		:host([size="small"]) .d2l-count-badge-number {
+			font-size: 0.6rem;
+			line-height: 0.9rem;
+			padding-left: 0.3rem;
+			padding-right: 0.3rem;
+		}
+
+		:host([size="large"]) .d2l-count-badge-number {
+			font-size: 0.8rem;
+			line-height: 1.2rem;
+			padding-left: 0.4rem;
+			padding-right: 0.4rem;
+		}
+
+		:host {
+			border: 2px white;
+			display: inline-block;
+			min-width: 0.9rem;
+		}
+
+		:host([size="small"]) {
+			border-radius: 0.45rem;
+		}
+
+		:host([size="large"]) {
+			border-radius: 0.6rem;
+		}
+
+		:host([type="notification"]) {
+			background-color: var(--d2l-color-carnelian-minus-1);
+		}
+
+		:host([type="count"]) {
+			background-color: var(--d2l-color-gypsum);
+		}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.type = 'count';
+		this.size = 'small';
+		this.hideZero = false;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.maxDigits && this.type === 'notification') {
+			// default to two digits for notification type
+			this.maxDigits = 2;
```

Comment:
I think this is fine to set here, because they shouldn't be changing `type` or `maxDigits` after the fact - and if they do, I don't think we want to try to adjust things on them.  Just calling this out in case someone has a use case I'm not thinking about.

Reply:
Late to the party, but another option would be to do this in the getter for the property. Se default it to `undefined` but then the getter returns `2` if the type is notification no matter what they set it to.